sympy/concrete/products.py

```diff
@@ -204,6 +204,17 @@ def _eval_is_zero(self):
         # a Product is zero only if its term is zero.
         return self.term.is_zero
 
+    def _eval_is_extended_real(self):
+        return self.function.is_extended_real
+
+    def _eval_is_positive(self):
+        if self.function.is_positive:
+            return True
```

Comment:
If this is an infinite product then can it not be zero or infinite even if the function is positive e.g.:
```julia
In [2]: Product(S(1)/2, (x, 1, oo)).doit()                                                                                        
Out[2]: 0

In [3]: Product(2, (x, 1, oo)).doit()                                                                                             
Out[3]: ∞
```
I think that a positive function can imply that the product is extended_nonnegative in the infinite case though.

Reply:
Makes sense. However if the limits are finite it shouldn't happen, right?

Reply:
It won't happen if the limits are finite. I think it would be good to add an `is_finite_product` property that can be used conveniently in places like this.

Reply:
I think `has_finite_limits` is my preferred name. Now, if there are no limits (as for Integrals, it makes sense to add this to `ExprWithLimits`), should one go for the assumptions on the variable or just return `None` if there are no explicit limits?

Btw, something similar happens for sums: `Sum(1/2, (x, 1, oo)).doit()` is `oo`.

Reply:
I think that `has_finite_limits` makes sense. In the context of an integral without limits it should be None I think. That probably makes the most sense since in most contexts you would use the property the None case would lead to an inconclusive result. (It probably also makes sense for Integrals to have something like an is_definite property if they don't)

Reply:
However, if you have a integration variable with is real, it should differ from extended_real. So right now, real leads to finite limits.